### PR TITLE
add task 2 - Learn Lodash: Compact

### PR DESCRIPTION
**Pull request by VolhaVaitulevich**
Task 2: https://edabit.com/challenge/M6fbYyBkzJXMAu39G

### CODE REVIEW

File: exercises/arrays/lodashCompact/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const arrTask2: any[] = [0, 1, false, 2, "", 3];
+
+const compact = (arrTask2: any[]): any => arrTask2.filter(Boolean);
+//return arr2.filter((value) => value);
```
Comment (siarhei-papkou) [inline exercises/arrays/lodashCompact/index.ts]: Please remove comented code

File: exercises/arrays/lodashCompact/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const arrTask2: any[] = [0, 1, false, 2, "", 3];
+
+const compact = (arrTask2: any[]): any => arrTask2.filter(Boolean);
```
Comment (siarhei-papkou) [inline exercises/arrays/lodashCompact/index.ts]: Try to improve typing here. 
https://www.typescriptlang.org/docs/handbook/2/generics.html
https://www.typescriptlang.org/docs/handbook/utility-types.html#omitthisparametertype

File: exercises/arrays/lodashCompact/index.ts (hunk 1 above)
Comment (VolhaVaitulevich) [reply exercises/arrays/lodashCompact/index.ts]: Done

File: exercises/arrays/lodashCompact/index.ts (hunk 2 above)
Comment (VolhaVaitulevich) [reply exercises/arrays/lodashCompact/index.ts]: I didn't completely understand what to do with `OmitThisParameter<Type>`, so I suggest adding here a new data type `MyArrayType` that will include types allowed for this array.

File: exercises/arrays/lodashCompact/index.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const arrTask2 = [0, 1, false, 2, "", 3];
+
+const compact = <T>(arrTask2: T[]): T[] => arrTask2.filter(Boolean);
```
Comment (siarhei-papkou) [inline exercises/arrays/lodashCompact/index.ts]: Let's handle falsy types properly
```ts
const compact = <T>(arrTask2: (T| FalsyValues)[]): T[] => arrTask2.filter((value): value is T =>Boolean(value));
```

Comment (VolhaVaitulevich) [reply exercises/arrays/lodashCompact/index.ts]: Made a new commit, added FalsyValues type

File: exercises/arrays/lodashCompact/index.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+type FalsyValues = 0 | null | false;
```
Comment (siarhei-papkou) [inline exercises/arrays/lodashCompact/index.ts]: Some values are  missed
https://developer.mozilla.org/en-US/docs/Glossary/Falsy#:~:text=A%20falsy%20(sometimes%20written%20falsey,such%20as%20conditionals%20and%20loops.&text=The%20keyword%20false%20.&text=The%20Number%20zero%20(so%2C%20also,etc.%2C%20and%200x0%20).

**Pull request merged**